pkg/sources/github/receive_adapter/receive_adapter.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+
+	webhooks "gopkg.in/go-playground/webhooks.v3"
+	"gopkg.in/go-playground/webhooks.v3/github"
+
+	"bytes"
+	"io/ioutil"
+	"net/http"
+
+	ghclient "github.com/google/go-github/github"
+	"log"
+)
+
+const (
+	// Target for messages
+	envTarget = "TARGET"
+	// Environment variable containing json credentials
+	envSecret = "GITHUB_SECRET"
+)
+
+// GithubHandler holds necessary objects for communicating with the Github.
+type GithubHandler struct {
+	client *ghclient.Client
+	ctx    context.Context
```

Comment:
This field doesn't seem to be used. Can it be removed?

Reply:
Removed.